### PR TITLE
Twitter api updates

### PR DESCRIPTION
**Pull request by stephen-lt-sl**
Changes the tweet search module to update at a greater rate, while also strictly adhering to Twitter api rate limits.

### CODE REVIEW

**Review by susanthomson — COMMENTED** [review]
The two gets have a fair bit of code repetition, is it worth pulling this out? Later we might have another one to get tweets from a specific list of users

**Comment by stephen-lt-sl** [comment]
Good catch; I didn't think it'd work well to move test cases to a shared function, but it does seem cleaner this way.

**Comment by susanthomson** [comment]
I meant in the actual server side code but tidying the tests is good too :joy:

**Comment by stephen-lt-sl** [comment]
This turned out to be a bit complicated, but I'm pretty sure it helped shrink the code a bit. Also, the repeated code inside the "addData" functions is somewhat likely to get changed on an individual basis in upcoming work on client updates, which is why it hasn't also been taken into a new function.

File: server/tweetSearch.js [hunk 1]
```diff
@@ -8,44 +8,54 @@ module.exports = function(client) {
...
+                    return statusA.id - statusB.id;
+                }));
+            }
         },
         "statuses/user_timeline": {
             since_id: 0,
             basePath: "statuses",
             requestsRemaining: 0,
             resetTime: 0,
+            addData: function(tweets) {
```
Comment (susanthomson) [inline server/tweetSearch.js:25]: Are lines 25-32 identical to 11-18?

Comment (stephen-lt-sl) [reply server/tweetSearch.js:25]: No, the tweets object has a "statuses" field in one and not in the other. As also mentioned in my comment above, the code that they do share might also not be shared for long.

Comment (susanthomson) [reply server/tweetSearch.js:25]: ah right :+1:

**Review by susanthomson — APPROVED** [review]

**Pull request merged**